lib/rules/max-params.js

```diff
@@ -61,25 +68,56 @@ module.exports = {
 		const sourceCode = context.sourceCode;
 		const option = context.options[0];
 		let numParams = 3;
+		let countVoidThis = false;
 
-		if (
-			typeof option === "object" &&
-			(Object.hasOwn(option, "maximum") || Object.hasOwn(option, "max"))
-		) {
-			numParams = option.maximum || option.max;
+		if (typeof option === "object") {
+			if (
+				Object.hasOwn(option, "maximum") ||
+				Object.hasOwn(option, "max")
+			) {
+				numParams = option.maximum || option.max;
+			}
+			countVoidThis = option.countVoidThis;
 		}
 		if (typeof option === "number") {
 			numParams = option;
 		}
 
+		/**
+		 * Removes 'this' parameter with void type annotation from parameter count
+		 * @param {ASTNode} node The node to process.
+		 * @returns {ASTNode} The processed node.
+		 * @private
+		 */
+		function removeVoidThisParam(node) {
+			if (
+				node.params.length === 0 ||
+				node.params[0].type !== "Identifier" ||
+				node.params[0].name !== "this" ||
+				node.params[0].typeAnnotation?.typeAnnotation.type !==
+					"TSVoidKeyword"
+			) {
+				return node;
+			}
+
+			return {
+				...node,
+				params: node.params.slice(1),
+			};
+		}
+
 		/**
 		 * Checks a function to see if it has too many parameters.
 		 * @param {ASTNode} node The node to check.
 		 * @returns {void}
 		 * @private
 		 */
 		function checkFunction(node) {
-			if (node.params.length > numParams) {
+			const processedNode = !countVoidThis
+				? removeVoidThisParam(node)
+				: node;
+
+			if (processedNode.params.length > numParams) {
```

Comment:
Can we simplify this? Creating a new node with different params seemed suitable for an extension rule to reuse the logic from the base rule. Here, if the function has a void this param, we could just compare its `params.length -1` instead of `params.length`.